spec/decorators/data_request_area_decorator_spec.rb

```diff
@@ -0,0 +1,121 @@
+require "rails_helper"
+
+describe DataRequestAreaDecorator, type: :model do
+  let(:data_request_area) { create(:data_request_area) }
+
+  describe "#location" do
+    context "without linked organisation" do
+      let(:decorated) { data_request_area.decorate }
+
+      it "uses location string" do
+        expect(decorated.location).to eq decorated.contact.name
+      end
+    end
+
+    context "with linked organisation" do
+      let(:contact) { create(:contact) }
+      let(:decorated) { data_request_area.decorate }
+
+      before do
+        data_request_area.update!(contact:)
+      end
+
+      it "uses name of organisation" do
+        expect(decorated.location).to eq decorated.contact.name
+      end
+    end
+  end
+
+  describe "#num_of_requests" do
+    let(:decorated) { data_request_area.decorate }
+
+    it "returns the count of data_requests" do
+      create(:data_request, data_request_area:)
```

Comment:
You can use [create_list](https://github.com/thoughtbot/factory_bot/blob/main/GETTING_STARTED.md#building-or-creating-multiple-records) to create multiple items